app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -57,10 +57,10 @@
     private PacketHandler mPacketHandler;
     private CommandsProto mCommandsProto;
     private AnalogConstants mAnalogConstants;
-    private I2C i2C;
-    private SPI spi;
-    private NRF24L01 nrf;
-    private MCP4728 dac;
+    public I2C i2C;
```

Comment:
Even I was thinking about it. Instead of context, i2c can be passed to the constructor.